### PR TITLE
Add blade directive @guest

### PR DESCRIPTION
**Pull request by LKaemmerling**
As addition to  https://github.com/laravel/framework/pull/20087 it would be nice if there is the opposite of @auth, so i implemented it.

### CODE REVIEW

**Comment by it-can** [comment]
If do not pass a guard, will this work?

**Comment by m1guelpf** [comment]
@it-can Yes

**Comment by LKDevelopment** [comment]
@it-can yeah it will, it is the same code as #20087. @m1guelpf what do you thing if the @guest can be an "else" for your @auth ? 

Like:

```blade
@auth

//Is authentificated

@guest 


// Is guest 

@endauth
```

**Comment by m1guelpf** [comment]
@LKDevelopment That'd be confusing. I'd suggest keeping both PR's as-is.

**Comment by it-can** [comment]
If I add a custom directive to my laravel 5.4 app, I get a error when calling 

AppServiceProvider.php
```
Blade::directive('auth', function ($guard = null) {
    return "<?php if(auth()->guard{$guard}->check()): ?>";
});
        
Blade::directive('endauth', function () {
    return "<?php endif; ?>";
});
```

This gives an error
```
@auth
    logged in
@endauth
```

this works...
```
@auth('web')
    logged in
@endauth
```

**Comment by m1guelpf** [comment]
@it-can You have a typo in the directive.
Try this:
``` php
Blade::directive('auth', function ($guard = null) {
    return "<?php if(auth()->guard({$guard})->check()): ?>";
});
        
Blade::directive('endauth', function () {
    return "<?php endif; ?>";
});
```

**Comment by it-can** [comment]
@m1guelpf yes that works... but when I check this PR, the code is missing the parenthese?

File: src/Illuminate/View/Compilers/Concerns/CompilesConditionals.php [hunk 1]
```diff
@@ -119,4 +119,25 @@ protected function compileEndAuth()
...
+ 
+    /**
+     * Compile the if-guest statements into valid PHP.
+     *
+     * @param  string|null  $guard
+     * @return string
+     */
+    protected function compileGuest($guard = null)
+    {
+        return "<?php if(auth()->guard{$guard}->guest()): ?>";
```
Comment (m1guelpf) [inline src/Illuminate/View/Compilers/Concerns/CompilesConditionals.php]: Should be `return "<?php if(auth()->guard({$guard})->guest()): ?>";`

**Comment by m1guelpf** [comment]
@LKDevelopment If you want, give me push access to your PR, and I'll apply the fix above and fix StyleCI & Travis :smile:

File: src/Illuminate/View/Compilers/Concerns/CompilesConditionals.php (hunk 1 above)
Comment (it-can) [inline src/Illuminate/View/Compilers/Concerns/CompilesConditionals.php]: should this also be changed in #20087  ?

Comment (m1guelpf) [inline src/Illuminate/View/Compilers/Concerns/CompilesConditionals.php]: athis is already changed in #20087

Comment (it-can) [inline src/Illuminate/View/Compilers/Concerns/CompilesConditionals.php]: Seems taylor changed it...

https://github.com/laravel/framework/commit/a11b44aeef5acdca514c8da26cb9c8d2596fc7ca

Comment (mathieutu) [inline src/Illuminate/View/Compilers/Concerns/CompilesConditionals.php]: No guys!
This is not the guard which is passed, but the complete $expression, with the ().
Nothing to change here!

Comment (m1guelpf) [inline src/Illuminate/View/Compilers/Concerns/CompilesConditionals.php]: @mathieutu is rigth.

File: src/Illuminate/View/Compilers/Concerns/CompilesConditionals.php [hunk 2]
```diff
@@ -119,4 +119,25 @@ protected function compileEndAuth()
     {
         return '<?php endif; ?>';
     }
+
```
Comment (m1guelpf) [inline src/Illuminate/View/Compilers/Concerns/CompilesConditionals.php]: Remove the extra whitespace here, and your PR is perfect!

**Comment by gabriel-caruso** [comment]
@m1guelpf @LKDevelopment Any ideas to work in @LKDevelopment's proposal?
```blade
@auth
    //Is authenticated
@guest 
    // Is guest 
@endauth
```

**Comment by m1guelpf** [comment]
@gabriel-caruso I don't think so. Now it'd be a breaking change...

**Pull request merged**